scipy/special/basic.py

```diff
@@ -2264,7 +2277,27 @@ def factorial(n, exact=False):
 
     """
     if exact:
-        return math.factorial(n)
+        if np.ndim(n) == 0:
+            return math.factorial(n)
+        else:
+            n = asarray(n)
+            un = np.unique(n)
+
+            if any(abs(un.astype(int)) != un):
```

Comment:
well they're converted to python int in line 2298 because if they're numpy int32 they'll wrap around instead of changing into long ints